### PR TITLE
C353545 Verify that Download matched records is hidden in case Errors only (firebird) (TaaS)

### PR DESCRIPTION
**Pull request by KarapetyanSofia**
[C353545](https://foliotest.testrail.io/index.php?/cases/view/353545)
[Allure Report](https://jenkins-aws.indexdata.com/job/folioRancher/job/folioTestingTools/job/runCypressTests/5199/allure)

image

### CODE REVIEW

File: cypress/e2e/bulk-edit/csv/bulk-edit-download-matched-records-is-hidden.cy.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import testTypes from '../../../support/dictionary/testTypes';
+import permissions from '../../../support/dictionary/permissions';
+import BulkEditSearchPane from '../../../support/fragments/bulk-edit/bulk-edit-search-pane';
+import devTeams from '../../../support/dictionary/devTeams';
+import getRandomPostfix from '../../../support/utils/stringTools';
+import FileManager from '../../../support/utils/fileManager';
+import Users from '../../../support/fragments/users/users';
+import BulkEditActions from '../../../support/fragments/bulk-edit/bulk-edit-actions';
+
+describe('Bulk Edit - Items', () => {
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/csv/bulk-edit-download-matched-records-is-hidden.cy.js:11]: pleas use nested describe blocks 
```
describe('bulk-edit', () => {
  describe('in-app approach', () => {

  });
});

```

File: cypress/e2e/bulk-edit/csv/bulk-edit-download-matched-records-is-hidden.cy.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      BulkEditSearchPane.selectRecordIdentifier('Item UUIDs');
+      BulkEditSearchPane.isDragAndDropAreaDisabled(false);
+      // Upload a .csv file with Items UUIDs (see Preconditions) by dragging it on the file drag and drop area
+      BulkEditSearchPane.uploadFile(userUUIDsFileName);
+      BulkEditSearchPane.waitFileUploading();
+      // Click the "Actions" menu
+      BulkEditActions.openActions();
+      BulkEditActions.downloadErrorsExists();
+      // Click the "Download errors (CSV)" button
+      BulkEditActions.downloadErrors();
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/csv/bulk-edit-download-matched-records-is-hidden.cy.js:53]: does this action download a file and are you cleaning up the file afterwards?

File: cypress/e2e/bulk-edit/csv/bulk-edit-download-matched-records-is-hidden.cy.js
Comment (nayimovag) [inline cypress/e2e/bulk-edit/csv/bulk-edit-download-matched-records-is-hidden.cy.js:1]: maybe move the file to `e2e/bulk-edit/in-app` folder and name it `bulk-edit-in-app-download-matched-records-is-hidden`

**Review by nayimovag — CHANGES_REQUESTED** [review]

File: cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-download-matched-records-is-hidden.cy.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        BulkEditSearchPane.isDragAndDropAreaDisabled(true);
+
+        BulkEditSearchPane.selectRecordIdentifier('Item UUIDs');
+        BulkEditSearchPane.isDragAndDropAreaDisabled(false);
+        // Upload a .csv file with Items UUIDs (see Preconditions) by dragging it on the file drag and drop area
+        BulkEditSearchPane.uploadFile(userUUIDsFileName);
+        BulkEditSearchPane.waitFileUploading();
+        // Click the "Actions" menu
+        BulkEditActions.openActions();
+        BulkEditActions.downloadErrorsExists();
```
Comment (nayimovag) [inline cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-download-matched-records-is-hidden.cy.js:54]: please also verify for `downloadMatchedRecordsAbsent`

Comment (KarapetyanSofia) [reply cypress/e2e/bulk-edit/in-app/bulk-edit-in-app-download-matched-records-is-hidden.cy.js:55]: Done!

**Review by nayimovag — APPROVED** [review]

**Review by Scorpibear — APPROVED** [review]

**Pull request merged**